crux-core/src/crux/codec.clj

```diff
@@ -164,6 +166,26 @@
   (-freeze-without-meta! [this out]
     (freeze-map out this)))
 
+(defn- biginteger->buffer [^BigInteger x type-id ^MutableDirectBuffer to]
+  (let [signum (.signum x)]
+    (-> to
+        (doto (.putByte 0 type-id))
```

Comment:
Was considering commenting on this stylistic repeated `doto` I noticed - and should've spotted last PR - I prefer a single `doto` for all the puts (see most older encoders), the combination of `->` and multiple `doto` essentially does the same thing.

Reply:
Sure - in this case it works, happy to change - in the others, it's combined with `mem/limit-buffer`. I'll have a play around with it, it might be that something like the following is easier to read:

```clojure
(-> (doto to
      ...)
    (mem/limit-buffer ...))
```

Failing that, we usually know the size of the buffer before putting to it, so this might also work:

```clojure
(doto (mem/limit-buffer to)
  ...)
```

Reply:
A glance at the ns feels like most places can be replaced by the first form, my general preference is that `doto` itself shouldn't sit repeated within the threading, a single one will do (with the puts nested beneath it), as it's also a kind of threading macro, but not of the return value.

Reply:
done @hraberg - cheers :)